drivers/sensor/grow_r502a/grow_r502a.h

```diff
@@ -0,0 +1,191 @@
+#define R502A_ACK_PACKET 0x7 /*Acknowledge packet*/
+#define R502A_END_DATA_PACKET 0x8 /*End of data packet*/
+
+/*Instruction code's definition*/
+#define R502A_GENIMAGE 0x01 /*Collect finger image*/
+#define R502A_IMAGE2TZ 0x02 /*To generate character file from image*/
+#define R502A_MATCH 0x03 /*Carry out precise matching of two templates*/
+#define R502A_SEARCH 0x04 /*Search the finger library*/
+#define R502A_REGMODEL 0x05 /*To combine character files and generate template*/
+#define R502A_STORE 0x06 /*To store template*/
+#define R502A_LOAD 0x07 /*To read/load template*/
+#define R502A_UPCHAR 0x08 /*To upload template*/
+#define R502A_DOWNCHAR 0x09 /*To download template*/
+#define R502A_IMGUPLOAD 0x0A /*To upload image*/
+#define R502A_DELETE 0x0C /*To delete template*/
+#define R502A_EMPTYLIBRARY 0x0D /*To empty the library*/
+#define R502A_SETSYSPARAM 0x0E /*To set system parameter*/
+#define R502A_READSYSPARAM 0x0F /*To read system parameter*/
+#define R502A_SETPASSWORD 0x12 /*To set password*/
+#define R502A_VERIFYPASSWORD 0x13 /*To verify password*/
+#define R502A_GETRANDOM 0x14 /*To generate a random code*/
+#define R502A_TEMPLATECOUNT 0x1D /*To read finger template numbers*/
+#define R502A_READTEMPLATEINDEX 0x1F /*Read fingerprint template index table*/
+#define R502A_LED_CONFIG 0x35 /*Aura LED Config*/
+#define R502A_CHECKSENSOR 0x36 /*Check sensor*/
+#define R502A_SOFTRESET 0x3D /*Soft reset*/
+#define R502A_HANDSHAKE 0x40 /*Handshake*/
+#define R502A_BADPACKET 0xFE /* Bad packet was sent*/
+
+#define R502A_STARTCODE 0xEF01 /*Fixed value, High byte transferred first*/
+#define R502A_DEFAULT_PASSWORD 0x00000000
+#define R502A_DEFAULT_ADDRESS 0xFFFFFFFF
+#define R502A_DEFAULT_CAPACITY 200
+#define R502A_HANDSHAKE_BYTE 0x55
+
+#define R502A_STARTCODE_IDX 0
+#define R502A_ADDRESS_IDX 2
+#define R502A_PKG_LEN_IDX 7
+#define R502A_PID_IDX 6 /* Package identifier index*/
+#define R502A_CC_IDX 9 /* Confirmation code index*/
+
+#define R502A_HEADER_LEN 9
+#define R502A_CHECKSUM_LEN 2 /* Checksum length in uart packages*/
+#define R502A_BUF_SIZE 368 /* Maximum Characteristic value size*/
+#define R502A_TEMPLATES_PER_PAGE 256
+#define R502A_TEMP_TABLE_BUF_SIZE	32
+#define R502A_NOFREEINDEX	-1
+
+#define DEFAULT_TIMEOUT 2000 /*UART reading timeout in ms*/
+#define R502A_DELAY 200
+
+#define LED_CTRL_BREATHING 0x01
+#define LED_CTRL_FLASHING 0x02
+#define LED_CTRL_ON_ALWAYS 0x03
+#define LED_CTRL_OFF_ALWAYS 0x04
+#define LED_CTRL_ON_GRADUALLY 0x05
+#define LED_CTRL_OFF_GRADUALLY 0x06
+
+#define LED_SPEED_HALF 0x50
+#define LED_SPEED_FULL 0xFF
+
+#define LED_COLOR_RED 0x01
+#define LED_COLOR_BLUE 0x02
+#define LED_COLOR_PURPLE 0x03
+
+struct led_params {
+	uint8_t color_code;
+	uint8_t color;
+	uint8_t speed; /* Speed 0x00-0xff */
+	uint8_t cycle; /* Number of cycles | 0-infinite, 1-255 */
+};
+
+struct buffer {
+	uint8_t data[R502A_BUF_SIZE];
+	uint8_t len;
```

Comment:
Oh, yes it can't have a larger value than 0xFF or 255 right? just now checked with the datasheet, and almost all the data packages used in the driver for sending and receiving packets are less than 17bytes only, so can I make this data buffer's size to 32?